src/OpenTelemetry.AutoInstrumentation/Loading/AspNetCoreDetector.cs

```diff
@@ -0,0 +1,48 @@
+// <copyright file="AspNetCoreDetector.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+
+namespace OpenTelemetry.AutoInstrumentation.Loading
+{
+    internal class AspNetCoreDetector : AssemblyLoadDetector
+    {
+        private static readonly IEnumerable<string> _requiredAssemblies = new string[]
+        {
+            "Microsoft.AspNetCore.Http"
+        };
+
+        public AspNetCoreDetector()
+            : base(_requiredAssemblies)
+        {
+        }
+
+        internal override Func<object> GetInstrumentationBuilder()
+        {
+            return () =>
+            {
+                var instrumentationType = Type.GetType("OpenTelemetry.Instrumentation.AspNetCore.AspNetCoreInstrumentation, OpenTelemetry.Instrumentation.AspNetCore");
+                var httpInListenerType = Type.GetType("OpenTelemetry.Instrumentation.AspNetCore.Implementation.HttpInListener, OpenTelemetry.Instrumentation.AspNetCore");
+
+                object httpInListener = Activator.CreateInstance(httpInListenerType, args: new OpenTelemetry.Instrumentation.AspNetCore.AspNetCoreInstrumentationOptions());
```

Comment:
I noticed that you're creating a new `AspNetCoreInstrumentationOptions` without reflection, but the other two types use reflection. Can we do it all without reflection?

Reply:
They are `internal` 😢

Reply:
Others are internal types in SDK. `AspNetCoreInstrumentationOptions` is actually pretty complicated type with funcs to filter and enrich. In case we would need to reconfigure options, it would be much easier to allocate it directly.